File: src/main/java/com/example/kanbansystem/entities/Task.java
package com.example.kanbansystem.entities;

import javax.persistence.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString(exclude = {"users", "sprints"})

@Entity
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    @Enumerated(EnumType.STRING)
    private TaskStatus status;
    private Integer estimation;
    @ManyToOne
    private Board board;
    @ManyToMany
    private List<User> users;
    @ManyToMany
    private List<Sprint> sprints;

}

File: src/main/kanbanSystem-front/front-web/kanban-web/src/app/models/Task.ts
import { Board } from "./Board";
import { User } from "./User";
import { TaskStatus } from "./TaskStatus";
import { Sprint } from "./Sprint";

export class Task {  
    constructor(
    public  id: number,
    public  name: string,
    public  description: string,
    public  status: TaskStatus,
    public  estimation: number,
    public  board: Board,
    public  users: User[],
    public  sprints: Sprint[]
    ) {}
  }
  
